### PR TITLE
HRfix Compatibility

### PR DESCRIPTION
**Pull request by seijihariki**
Compatibility fix with the old HRFix versions of the webui using the original openOutpaint HRFix implementation (firstphase_width = width / 2 and firstphase_height = height / 2).

Still displays alert on boot.

### CODE REVIEW

File: js/ui/tool/dream.js [hunk 1]
```diff
@@ -141,41 +141,7 @@ const _dream = async (endpoint, request) => {
 	let data = null;
 	try {
 		generating(true);
-		if (
```
Comment (seijihariki) [inline js/ui/tool/dream.js:144]: This was moved to clean up _dream() a little bit. As we only use txt2img in one case (dream tool when nothing is under the cursor), moving it there seems better.

File: js/ui/tool/dream.js [hunk 2]
```diff
@@ -849,6 +815,50 @@ const dream_generate_callback = async (bb, resolution, state) => {
...
+		// For compatibility with the old HRFix API
+		if (global.isOldHRFix && request.enable_hr) {
+			request.firstphase_width = request.width / 2;
+			request.firstphase_height = request.height / 2;
+		}
+
+		// Only set this if HRFix is enabled in the first place
+		request.denoising_strength = request.enable_hr
+			? stableDiffusionData.hr_denoising_strength
+			: 1;
```
Comment (seijihariki) [inline js/ui/tool/dream.js:860]: Also, not sure if this is required at all on old HRFix. If not, `(!global.isOldHRFix && request.enable_hr)` should replace here.

Comment (zero01101) [reply js/ui/tool/dream.js:860]: i wasn't sure if that was necessary or not but figured better safe than sorry lol

**Comment by zero01101** [comment]
i've got an outdated version of SD on a different machine with a... very very very very slow tesla m40 card in it lol, testing this now

File: js/ui/tool/dream.js (hunk 1 above)
Comment (zero01101) [reply js/ui/tool/dream.js:144]: totally makes sense, i guarantee i was _scrambling_ to get something workable pushed out lol, i predicted HAVOC when i saw that A1111 commit ;)

**Comment by zero01101** [comment]
image

😚 🤌 _magnifique_

**Review by zero01101 — APPROVED** [review]
you're a hero and i owe you a beer or a meal or maybe a new car

**Comment by zero01101** [comment]
also if you were curious about _how_ slow a tesla m40 24gb card is at HRfixing a 512x512 to 1024x1024 lmao

image

File: js/ui/tool/dream.js (hunk 2 above)
Comment (seijihariki) [reply js/ui/tool/dream.js:860]: Haha makes sense! Now that I see it, hr_denoising_strength seems to be a new option in the new HRFix, right? Will just replace this and merge to main.

**Comment by seijihariki** [comment]
Oh, 2 minutes is quite painful indeed. Will be merging this into main then!

**Comment by zero01101** [comment]
AND THAT WAS FOR BATCH/ITER SIZE 1 D: D:

**Pull request merged**